engine-tests/src/test/java/org/terasology/metatesting/EntitySystemTest.java

```diff
@@ -0,0 +1,292 @@
+package org.terasology.metatesting;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.terasology.engine.context.Context;
+import org.terasology.engine.context.internal.ContextImpl;
+import org.terasology.engine.core.ComponentSystemManager;
+import org.terasology.engine.core.module.ModuleManager;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.entity.internal.EngineEntityManager;
+import org.terasology.engine.entitySystem.entity.internal.PojoEntityManager;
+import org.terasology.engine.entitySystem.event.AbstractConsumableEvent;
+import org.terasology.engine.entitySystem.event.EventPriority;
+import org.terasology.engine.entitySystem.event.Priority;
+import org.terasology.engine.entitySystem.event.internal.EventSystem;
+import org.terasology.engine.entitySystem.event.internal.EventSystemImpl;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.ComponentSystem;
+import org.terasology.engine.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.engine.network.NetworkMode;
+import org.terasology.engine.network.NetworkSystem;
+import org.terasology.engine.recording.RecordAndReplayCurrentStatus;
+import org.terasology.engine.registry.CoreRegistry;
+import org.terasology.engine.registry.In;
+import org.terasology.gestalt.entitysystem.component.Component;
+import org.terasology.gestalt.entitysystem.event.Event;
+import org.terasology.gestalt.entitysystem.event.ReceiveEvent;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class EntitySystemTest {
+
+    private EngineEntityManager entityManager;
+    private EventSystem eventSystem;
+    private Context context;
+
+    @BeforeEach
+    public void setUp() {
+        context = new ContextImpl();
+        CoreRegistry.setContext(context);
```

Comment:
I am sorely tempted to put an apology in code everwhere we have to populate `CoreRegistry`. Either way, can populating the registry be changed to use the new immutable variant? You will need to pass a `ServiceRegistry` into the constructor for this.

It is really useful to use an immutable context in unit tests as it allows us to catch out the odd cases where classes are still randomly writing to and overwriting the context where they should not be.

Reply:
Sure, got an example handy? I can perhaps find one in the big PR when I have time to hunt it down